In [1]:
import numpy as np
from sklearn import datasets
from sklearn.manifold import MDS
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
from sklearn.model_selection import cross_val_score

# 1. Load the IRIS dataset
iris = datasets.load_iris()
X, y = iris.data, iris.target

# 2. LDA classification on the original dataset
lda_original = LDA()
scores_original = cross_val_score(lda_original, X, y, cv=5)
mean_original = np.mean(scores_original)
print("Accuracy on original features: {:.2f}%".format(mean_original * 100))

# 3. Dimensionality reduction using MDS (reducing to 2 dimensions)
mds = MDS(n_components=2, random_state=42)
X_mds = mds.fit_transform(X)

# 4. LDA classification on the MDS-reduced dataset
lda_mds = LDA()
scores_mds = cross_val_score(lda_mds, X_mds, y, cv=5)
mean_mds = np.mean(scores_mds)
print("Accuracy on MDS-reduced features: {:.2f}%".format(mean_mds * 100))


Accuracy on original features: 98.00%
Accuracy on MDS-reduced features: 96.00%


Full Features (98%): All discriminative information is retained, leading to excellent classification.<br>
MDS-Reduced (96%): Slight information loss occurs during dimensionality reduction, causing a small drop in accuracy.